addons/html_editor/static/src/main/power_buttons_plugin.js

```diff
@@ -0,0 +1,124 @@
+import { Plugin } from "@html_editor/plugin";
+import { closestBlock } from "@html_editor/utils/blocks";
+import { isEmptyBlock } from "@html_editor/utils/dom_info";
+import { closestElement } from "@html_editor/utils/dom_traversal";
+import { _t } from "@web/core/l10n/translation";
+
+export class PowerButtonsPlugin extends Plugin {
+    static name = "power_buttons";
+    static dependencies = ["selection", "local-overlay", "powerbox"];
+    /** @type { (p: PowerButtonsPlugin) => Record<string, any> } */
+    static resources = (p) => ({
+        layoutGeometryChange: p.updatePowerButtons.bind(p),
+        onSelectionChange: p.updatePowerButtons.bind(p),
+        powerButtons: [
+            {
+                id: "more_options",
+                name: _t("More options"),
+                fontawesome: "fa-ellipsis-v",
+                action: () => {
+                    p.openPowerbox();
+                },
+            },
+        ],
+    });
```

Comment:
Since the type of `powerButtons` is now `Array<String>`, I think this should be moved to `setup`. It's just the extra button that is part of the UI. It can be hardcoded IMO and does not need to be a resource. That way you can remove the `|| id` in the filter which is actually kind of confusing because the only time it is useful is when `id` is _not_ an id but this object here... ^^'